### ISSUE TITLE
Valor Total da Compra aparece dobrado no checkout

### ISSUE THREAD

**Issue by lgustavonogueira** [deivisonarthur/OSC-Magento-Brasil-6-Pro#206]
Olá,

Primeiramente parabéns pelo trabalho de todos os envolvidos, excelente módulo.

Estou com um problema e não encontrei outros devs com esse issue. No meu carrinho de compras o valor total (Gran Total em inglês) aparece normalmente, quando avanço para página de checkout o valor dobra (de $400 para $800, por exemplo). Ao finalizar a compra, o valor mostrado no comprovante de pedido e no histórico de transações é o de $400 (ou valor esperado).

Alguém já enfrentou esse problema? Uso Magento 1.9.2.2 e OSC 6 em um tema pago (SuperTheme EM 0132).

![osc issue 01](https://cloud.githubusercontent.com/assets/16902607/13825134/e8cbbc9a-eb8f-11e5-963d-15c3a60c2442.png)
![osc issue 02](https://cloud.githubusercontent.com/assets/16902607/13825136/e8d3f4e6-eb8f-11e5-9888-563b8e347791.png)
![osc issue 03](https://cloud.githubusercontent.com/assets/16902607/13825135/e8d195d4-eb8f-11e5-9f5a-846d4212f54c.png)

**Comment by rafaelstz** [comment]
Está com a última versão? Se trocar o tema para o RWD pelo admin, o erro persiste?

**Comment by lgustavonogueira** [comment]
Sim, é a versão corrente do Guthub que utilizo. Troquei para o tema padrão magento e agora o Valor Total aparece zerado (R$ 0,00)

Fiz um teste com a versão 4 do OSC e o problema se resolve (apesar de novos bugs em outros campos aparecerem)


![osc issue 04](https://cloud.githubusercontent.com/assets/16902607/13850922/d538f08e-ec3a-11e5-9a6e-19dcc03d3ece.png)


Estou pesquisando e aparentemente o bug pode estar relacionado com dados duplicados na tabela sales_flat_quote_address

**Comment by isaaclopes** [comment]
Ola, @lgustavonogueira 
tenta limpar o quote "NO AMBIENTE DEV" para ver se o erro persiste

no seu "DEV" raiz do seu magento coloca o arquivo https://gist.githubusercontent.com/isaaclopes/b4af82364fbefae7af9b/raw/6ee9a97215b09827e3ef205f50bdc8ca926e8ba6/clear_quote.php
ele vai limpar todos os quote/carrinho.

não faz em "produção" horario de venda pois ele limpa o carrinho dos clientes.

**Comment by StarTron** [comment]
Srs, gostaria de reforçar os parabéns pelo excelente trabalho de todos os envolvidos neste módulo.

Estou com o mesmo problema, mas no meu caso estou utilizando um tema da emthemes.com.
Complementando o que o já foi informado aqui: 
- o **clear_quote.php**, cumpre o papel dele, mas infelizmente não teve o resultado esperado em relação ao valor total.  
- Testei nos demais temas default e o total também aparece com o valor zerado, similar ao que o  **lgustavonogueira** postou.

Mas aqui para mim teve um detalhe que fez toda diferença. 

Para produtos virtuais, onde não existe à necessidade de **utilizar o frete** (shipping method), independente do tema e quantidade de itens no carrinho o Total Geral aparece com o **valor correto**. Caso inclua algum item que necessite utilizar o frete o valor total aparece em dobro no checkout.
Também fiz testes utilizando regras de promoções e obtive os mesmos resultados.

Alguém conseguiu alguma solução para esse "detalhe" do total geral? 
Existe alguma configuração que deva ser feita para evitar esse erro?

Uso Magento 1.9.1.0 e OSC 6 (não testei outras versões do OSC)

**Comment by hansbonini** [comment]
Faça uma instalação apenas com o Maento e o OSC. Depois vá adicionando os módulos e verificando qual está entrando em conflito.

**Comment by kellerkichel** [comment]
Mesmo problema aqui, magento instalação clean 1.9.2 + OSC e o valor vem zerado.
Um detalhe que ocorre aqui é que quando mudo a forma de entrega as opções de pagamento desaparecem da tela.

Alguma dica ou novidade neste parte?

**Comment by kellerkichel** [comment]
Pessoal, minutos depois de comentar que estava com o problema conversei com um colega meu aqui do escritório e ele achou o problema, esta no arquivo /app/code/community/Inovarti/Onestepcheckout/controllers/IndexController.php nas linhas 63 e 64, as linha estão comentadas, este comentário foi feito no commit https://github.com/deivisonarthur/OSC-Magento-Brasil-6-Pro/commit/e4a1e4c41d1fd0e12f1d310ea1e0ddafbaaddeea#diff-7757397e57287fc68e3ada8fac2c1af3 não sei se tem algum motivi especifico, mas descomentando estas linhas tudo voltou ao normal.

![screenshot_1](https://cloud.githubusercontent.com/assets/4204423/17001468/8275b92a-4e9d-11e6-8243-547aa6e9d746.png)

**Comment by StarTron** [comment]
Srs, grato pela colaboração. Acredito que o problema esta no tema da emthemes.com mesmo.

Seguindo as orientações postadas, mesmo numa instalação limpa do Magento e OSC, os resultados foram similares ao do kellerkichel, também retirei os comentários nas linhas (63 - 64) e realmente o normalizou.  
Mas ao adicionar o tema da emthemes.com, o carrinho continuou apresentando o mesmo problema inicialmente relatado (total dobrado). Resta tentar identificar o foco do problema e caso o encontre posto aqui.
Grato à todos.

**Comment by hansbonini** [comment]
Como você já isolou o problema fica mais fácil detectar, esse tema tem carrinho AJAX?

**Comment by StarTron** [comment]
Olá hansbonini, sim o tema tem carrinho AJAX. 
Eu comuniquei o desenvolvedor (suporte da emthemes) sobre esse problema e ainda estou esperando uma resposta (posto aqui os resultados).
Imaginei que uma solução simples seria suprimir o carrinho no checkout (fiz isso e aparentemente funciona) , mas sei que essa não é uma solução adequada.

**Comment by hansbonini** [comment]
Bem provável que o problema se encontre no controller Ajax do tema.

**Comment by gumaloko** [comment]
Na pasta checkout/cart/ há um arquivo com o nome sidebar.phtml dentro do do tema da EM Themes

Dentro do arquivo basta comentar a seguinte linha:

$quote->getShippingAddress()->collectTotals();

E pronto, valor dobrado desaparece do checkout e aparece o valor correto.
